default.nix

```diff
@@ -0,0 +1,251 @@
+      (shareTable 11 28 12 14)
+      (shareTable 6 4 2 15)
+      (shareTable 10 17 21 20)
+      (shareTable 26 30 7 5)
+      recovery
+      fusionInner
+      fusionOuter
+      checksumTable1
+      checksumTable2
+      checksumWorksheet
+    ];
+  };
+
+  dependencyContentRecur = content: builtins.concatMap
+    (item: (dependencyContentRecur item.dependencies) ++ [ item.content ])
+    content;
+  dependencyContent = pages: lib.lists.unique (
+    (map (dep: dep.content) standardDependencies) ++
+    (builtins.concatMap (page: dependencyContentRecur page.dependencies) pages)
+  );
+
+  renderBooklet = booklet:
+    let
+      addPage = content: pageData: {
+        content = content.content + lib.optionalString (pageData ? sourceHeader) ''
+          %****************************************************************
+          %*
+          %* ${pageData.sourceHeader}
+          %*
+          %****************************************************************
+        '' + ''
+          %%Page: ${toString content.nextPgIdx} ${toString content.nextPgIdx}
+          ${lib.optionalString (pageData ? isLandscape) "%%PageOrientation: Landscape\n"}
+          %%BeginPageSetup
+          /pgsave save def
+          %%EndPageSetup
+          ${if pageData ? isLandscape then "landscapePage" else "portraitPage"} begin
+          ${lib.optionalString (pageData ? drawFooter) "${toString content.nextFooterIdx} drawFooter"}
+          ${pageData.content}
+          end
+          pgsave restore
+          showpage
+        '';
+        nextFooterIdx = content.nextFooterIdx + (if pageData ? drawFooter then 1 else 0);
+        nextPgIdx = content.nextPgIdx + 1;
+      };
+      initialContent = {
+        content = ''
+          %!PS-Adobe-3.0
+          %%Orientation: Portrait
+          %%Pages: ${toString (builtins.length booklet.pages)}
+          %%EndComments
+          %%BeginSetup
+          ${toString (dependencyContent (booklet.pages))}%%EndSetup
+
+          %************************************************************************
+          %************************************************************************
+          %*
+          %* Section Three: Page Rendering
+          %*
+          %************************************************************************
+          %************************************************************************
+
+        '';
+        nextPgIdx = 1;
+        nextFooterIdx = 1;
+      };
+      finalContent = builtins.foldl' addPage initialContent booklet.pages;
+    in
+    pkgs.writeTextFile {
+      name = booklet.name;
+      text = finalContent.content + ''
+        %%EOF
+      '';
+    };
+  checkSinglePage = page: renderBooklet {
+    name = "test-single-page.ps";
+    pages = [ page ];
+  };
+in
+stdenv.mkDerivation {
+  name = "codex32${shortId}";
+
+  nativeBuildInputs = if doPdfGeneration then [ ghostscript ] else [ ];
+
+  phases = [ "buildPhase" ] ++ lib.optionals doCheck [ "checkPhase" ];
```

Comment:
Unfortunately the `checkPhase` is never run because `doCheck` needs to be set to `true` in the `stdenv.mkDerivation` in order to run the `checkPhase`

You should change the phases line to:

    phases = [ "buildPhase" "checkPhase" ];
    inherit doCheck;

and let `doCheck` control whether or not the check phase is executed or not.